src/core/opamSystem.ml

```diff
@@ -1633,6 +1633,40 @@ let translate_patch ~dir orig corrected =
   end;
   close_in ch
 
+let gpatch = lazy begin
+  let rec search_gpatch = function
+    | [] -> None
+    | patch_cmd::patch_cmds ->
+      match OpamProcess.run (make_command ~name:"patch" patch_cmd ["--version"]) with
+      | r ->
+        (match OpamProcess.is_success r, r.OpamProcess.r_stdout with
+         | true, full::_ when
+             OpamStd.String.is_prefix_of ~from:0 ~full "GNU patch " ->
+           Some patch_cmd
+         | _ ->
+           search_gpatch patch_cmds)
+      | exception _ -> search_gpatch patch_cmds
+  in
+  let default_cmd, other_cmds =
```

Comment:
Init defaults is not aligned with this selection
For the moment, it is
```
required-tools: [
 "patch" {} {!(os = "openbsd" | os = "freebsd")}
 "gpatch" {} {os = "openbsd" | os = "freebsd"}
...
```
Shouldn't we add DragonFly and NetBSD, or change to `os-family = "bsd"` ?